### PR TITLE
[Frontend][Review Metrics] Review Metrics Playtesting follow-ups

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

- Fix overwritten values count
  - Due to duplicate datapoint values in the Data Upload API response, we overcount overwritten values sometimes. 
    - I filed a task to fix this on the [backend](https://github.com/Recidiviz/justice-counts/issues/31), but for now, we can ignore duplicates in the frontend. I didn't put a TODO to undo this change because I don't think it's really necessary. 
- Change "Overwrites" to "Overwritten Values"
- Allow continuing to the Review Metrics page even if the upload had errors
Screen Shot 2022-09-21 at 9 26 14 AM
- Tweaked Review Metrics to make it more clear that their data has already been successfully uploaded
- Changed the "Review 0 Uploaded Metrics" copy into a very simple No Metrics to Review
Screen Shot 2022-09-21 at 9 59 32 AM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/DataUpload/UploadErrorsWarnings.tsx [hunk 1]
```diff
@@ -154,8 +154,7 @@ export const UploadErrorsWarnings: React.FC<UploadErrorsWarningsProps> = ({
                     removeSnakeCase(selectedSystem).toLowerCase()}
                 </a>
               </span>{" "}
-              system. To continue, please resolve the errors in your file and
-              reupload.
+              system.
```
Comment (lilidworkin) [inline publisher/src/components/DataUpload/UploadErrorsWarnings.tsx:157]: Nice, good call on removing this for now!

File: publisher/src/components/DataUpload/UploadErrorsWarnings.tsx [hunk 2]
```diff
@@ -164,19 +163,16 @@ export const UploadErrorsWarnings: React.FC<UploadErrorsWarningsProps> = ({
...
-                navigate("/review-metrics", {
-                  state: metrics,
-                  replace: true,
-                })
-              }
-            >
-              Continue
-            </Button>
-          )}
+          <Button
```
Comment (lilidworkin) [inline publisher/src/components/DataUpload/UploadErrorsWarnings.tsx:166]: Can you add a screenshot showing what this page looks like now if there are errors? I think there will be two calls to action -- a blue button on the left that says New Upload and a blue button on the right that says Continue. Is that right? That's fine for now (we need input on product and design on this) but just want to make sure I know the current behavior. Thanks!

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx [hunk 3]
```diff
@@ -137,7 +146,7 @@ const ReviewMetrics: React.FC = observer(() => {
           <SectionTitle>{metric.display_name}</SectionTitle>
           {overwrittenValuesCount > 0 && (
             <SectionTitleOverwrites>
-              * {overwrittenValuesCount} Overwrite
+              * {overwrittenValuesCount} Overwritten Value
```
Comment (lilidworkin) [inline publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:149]: Nice, much better!

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx [hunk 4]
```diff
@@ -264,20 +280,18 @@ const ReviewMetrics: React.FC = observer(() => {
...
           <a href="mailto:support@justice-counts.org">
             support@justicecounts.org
           </a>
         </Subheading>
         {filteredMetrics.map((metric, idx) => {
-          if (metric.datapoints.length > 0) {
-            return renderSection(metric, idx);
-          }
-          return null;
+          return renderSection(metric, idx);
```
Comment (lilidworkin) [inline publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:294]: What does this change do?

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx [hunk 5]
```diff
@@ -248,6 +245,25 @@ const ReviewMetrics: React.FC = observer(() => {
...
+    value: string | number | null,
+    oldValue: string | number | null
+  ) => {
+    if (value === null) {
+      return null;
+    }
+    return (
+      <DatapointsTableDetailsCell key={key}>
+        {parseFloat(
+          (typeof value === "string" ? parseFloat(value) : value).toFixed(2)
```
Comment (lilidworkin) [inline publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:259]: Actually not sure we want to do this -- I just commented on the doc, the issue with the "too many sig figs" was that what we were showing in the UI didn't match what was in the spreadsheet.  I think if there are 2+ decimals in the spreadsheet, it's fine to show in the UI without rounding. LMK if that makes sense!

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx [hunk 6]
```diff
@@ -106,9 +106,6 @@ const ReviewMetrics: React.FC = observer(() => {
     }, {} as { [key: string]: number });
 
     metric.datapoints.forEach((dp) => {
-      if (dp.old_value !== null) {
```
Comment (lilidworkin) [inline publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:109]: Thanks for this fix!! Are the duplicate values on the backend causing any other issues? We should definitely fix the root cause on the backend, just curious if there are any other patches we need to do on the frontend in the meantime.

**Review by lilidworkin — COMMENTED** [review]
Nice, thanks for knocking out all these fixes @terryttsai ! Appreciate how you enumerated everything that this PR does in the description too. 🙌

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 4 above)
Comment (terryttsai) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:294]: Nothing, this is just something I meant to do earlier from an earlier fix PR!

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 5 above)
Comment (terryttsai) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:259]: I'll remove this, I'd rather accurately portray what the user gave us, if it's showing up as too many sig figs from the backend, I'll leave it up to Nichelle to fix!

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 6 above)
Comment (terryttsai) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:109]: Yep, this is the issue to fix: https://github.com/Recidiviz/justice-counts/issues/31, and on the frontend, I'm rooting out duplicate datapoints before using it and hadn't found any more issues with this.

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 5 above)
Comment (lilidworkin) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:259]: I'm not actually sure where it's showing up as too many sig figs though -- could be backend or could be frontend -- can you coordinate with @nichelle-hall to investigate?

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 6 above)
Comment (lilidworkin) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:109]: Sounds good!

File: publisher/src/components/DataUpload/UploadErrorsWarnings.tsx (hunk 2 above)
Comment (terryttsai) [reply publisher/src/components/DataUpload/UploadErrorsWarnings.tsx:166]: Added screenshot in the description

Comment (terryttsai) [reply publisher/src/components/DataUpload/UploadErrorsWarnings.tsx:166]: Only the button on the right is blue

File: publisher/src/components/ReviewMetrics/ReviewMetrics.tsx (hunk 5 above)
Comment (terryttsai) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx:259]: This is coming from the backend

Comment (terryttsai) [reply publisher/src/components/ReviewMetrics/ReviewMetrics.tsx]: Removed logic to round decimals

**Comment by terryttsai** [comment]
if my math is right, (6/16)^9 or 0.015% chance of a commit hash starting with 9 letters in a row
https://github.com/Recidiviz/justice-counts/pull/32/commits/eccdddcea466766f38349773c35ed5fb1af23cec

**Comment by terryttsai** [comment]
Added a very simple No Metrics to Review state for the review metrics page

**Review by lilidworkin — APPROVED** [review]
Awesome!!

**Pull request merged**